Automattic/vip-go-mu-plugins
# Search: Add --mode to validate-contents command

## Description
To assist with debugging, it'd be nice to have a `--mode` flag to the `wp vip-search health validate-contents` command to limit to a subset of inconsistencies.

This PR also updates the reporting term to use "mismatch" instead of "inconsistency" for posts that exist but have differing content.

## Changelog Description

### Plugin Updated: Enterprise Search

Added --mode command options to validate-contents command.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1) Create a few posts
2) Delete one from the DB:
```
wp db cli
> DELETE FROM wp_posts WHERE id = <postid>
```
3) Modify the post_content for another in the DB:
```
wp db cli
> UPDATE wp_posts SET post_content = 'blahblahblah' WHERE id = <another-postid>
```
4) Run `wp vip-search health validate-contents --do_not_heal` and expect to see one marked as `extra_in_index` and another marked as `inconsistent`
5) Apply PR

For the `missing` input:
6) Run `wp vip-search health validate-contents --mode=missing --do_not_heal` and expect to see only the one marked as `extra_in_index` reported back

For the `--mismatch` flag:
7) Run `wp vip-search health validate-contents --mode=mismatch --do_not_heal` and expect to see only the one marked as `mismatch` reported back
